tests/lib/rules/no-else-return.js

```diff
@@ -182,6 +182,24 @@ ruleTester.run("no-else-return", rule, {
             output: "function foo21() { var x = true; if (x) { return x; } if (x === false) { return false; } }",
             options: [{ allowElseIf: false }],
             errors: [{ messageId: "unexpected", type: "IfStatement" }]
+        },
+        {
+            code: "function foo() { if (true) { return bar; } else { const baz = 1;  return baz; } }",
+            output: null,
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ messageId: "unexpected", type: "BlockStatement" }]
+        },
+        {
+            code: "function foo() { if (true) { return bar; } else { let baz = 1;  return baz; } }",
+            output: null,
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ messageId: "unexpected", type: "BlockStatement" }]
+        },
+        {
+            code: "function foo() { let bar = true; if (baz) { return; } else { let bar = false;  return; } }",
```

Comment:
can you add a test case with the following code:

```js
function foo() {
  let bar = true;
  if (baz) {
    return;
  } else {
    {
      let bar = false; 
    }
    return;
  }
}
```
This code is able to be fixed as the block scoped declaration is contained in another block. I think your current implementation just returns `null`

Reply:
@ajafff whats the output supposed to be in this example?

```
function foo() {
  let bar = true;
  if (baz) {
    return;
  } else {
    {
      let bar = false; 
    }
  }
}
```